src/SaveHandler/Cache.php

```diff
@@ -2,57 +2,25 @@
 
 namespace Laminas\Session\SaveHandler;
 
-use Laminas\Cache\Storage\ClearExpiredInterface as ClearExpiredCacheStorage;
-use Laminas\Cache\Storage\StorageInterface as CacheStorage;
-use ReturnTypeWillChange;
+use Psr\SimpleCache\CacheInterface;
 
 /**
  * Cache session save handler
- *
- * @see ReturnTypeWillChange
  */
-class Cache implements SaveHandlerInterface
+final class Cache implements SaveHandlerInterface
 {
-    /**
-     * Session Save Path
-     *
-     * @var string
-     */
-    protected $sessionSavePath;
-
-    /**
-     * Session Name
-     *
-     * @var string
-     */
-    protected $sessionName;
-
-    /**
-     * The cache storage
-     *
-     * @var CacheStorage
-     */
-    protected $cacheStorage;
-
-    /**
-     * Constructor
-     */
-    public function __construct(CacheStorage $cacheStorage)
-    {
-        $this->setCacheStorage($cacheStorage);
+    public function __construct(
+        private CacheInterface $cacheStorage,
+        private string $sessionSavePath,
+        private string $sessionName
+    ) {
     }
 
     /**
      * Open Session
-     *
-     * @param string $path
-     * @param string $name
-     * @return bool
      */
-    #[ReturnTypeWillChange]
-    public function open($path, $name)
+    public function open(string $path, string $name): bool
     {
-        // @todo figure out if we want to use these
```

Comment:
Not sure this method needs to do anything other than `return true` - there is no "open" step, we just read and write to the cache